src/topology/homotopy/casts.lean

```diff
@@ -0,0 +1,79 @@
+/-
+Copyright (c) 2022 Praneeth Kolichala. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Praneeth Kolichala
+-/
+import topology.homotopy.fundamental_groupoid
+
+
+/-!
+# Fundamental group casts
+
+This file defines `hcast`, which is just an abbreviation for `eq_to_hom` for the
+fundamental groupoid. In particular, it accepts an equality `x₀ = x₁` and returns
+an arrow from `x₀` to `x₁`, where `x₀` and `x₁` are points in some topological space.
+
+
+
+-/
+noncomputable theory
+
+namespace fundamental_groupoid_functor
+universe u
+open fundamental_groupoid
+
+local attribute [instance] path.homotopic.setoid
+
+section
+
+variables {X : Top} {x₀ x₁ x₀' x₁' : X} (p : path x₀ x₁)
+
+/-- Abbreviation for `eq_to_hom` that accepts points in a topological space -/
+abbreviation hcast (hx : x₀ = x₁) : from_top x₀ ⟶ x₁ := category_theory.eq_to_hom hx
```

Comment:
Having `from_top` on the domain but not on the codomain seems quite asymmetrical to me.

Reply:
Since `from_top` is marked reducible and equal to the identity, it shouldn't matter at all proof-wise. Aesthetically, I'm fine with either, but I've used `from_top a ⟶ b` in other places as well where `a` and `b` are points in a topological space, just because it's shorter (the reason putting only one works is because once Lean realizes the first argument of `⟶`, namely `from_top a`, is an object from the fundamental groupoid of `a`, it is looking for another object in the fundamental  groupoid as the second argument, and it can reduce that to a point in the topological space).

Reply:
I agree with Johan's comment here.